File: server/src/main/java/com/edutech/progressive/entity/Enrollment.java
package com.edutech.progressive.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import javax.persistence.*;
import java.util.Date;

@Entity
@Table(name = "enrollment", uniqueConstraints = {
        @UniqueConstraint(columnNames = {"student_id", "course_id"})
})
public class Enrollment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "enrollment_id")
    private int enrollmentId;

    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "student_id", nullable = false)
    private Student student;

    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "course_id", nullable = false)
    private Course course;

    @Temporal(TemporalType.DATE)
    @Column(name = "enrollment_date", nullable = false)
    private Date enrollmentDate;

    public Enrollment() {}

    public Enrollment(Student student, Course course, Date enrollmentDate) {
        this.student = student;
        this.course = course;
        this.enrollmentDate = enrollmentDate;
    }

    public int getEnrollmentId() { return enrollmentId; }
    public Student getStudent() { return student; }
    public Course getCourse() { return course; }
    public Date getEnrollmentDate() { return enrollmentDate; }

    public void setEnrollmentId(int enrollmentId) { this.enrollmentId = enrollmentId; }
    public void setStudent(Student student) { this.student = student; }
    public void setCourse(Course course) { this.course = course; }
    public void setEnrollmentDate(Date enrollmentDate) { this.enrollmentDate = enrollmentDate; }
}

File: client/src/app/educonnect/models/Enrollment.ts
<<<<<<< HEAD
export class Enrollment {
  enrollmentId: number;
  studentId: number;
  courseId: number;
  enrollmentDate: Date;
  constructor(enrollmentId: number, studentId: number, courseId: number, enrollmentDate: Date) {
    this.enrollmentId = enrollmentId;
    this.studentId = studentId;
    this.courseId = courseId;
    this.enrollmentDate = enrollmentDate;
  }
  logAttributes(): void {
    console.log('enrollmentId:', this.enrollmentId);
    console.log('studentId:', this.studentId);
    console.log('courseId:', this.courseId);
    console.log('enrollmentDate:', this.enrollmentDate);
  }
}
=======

export class Enrollment {

}

>>>>>>> d76e0db293a8626f50b912ae8482fe15a23e1abc
